File: solutions/javascript/difference-of-squares/1/difference-of-squares.js
export class Squares {
	constructor(n) {
		this.n = n;
	}

	get sumOfSquares() {
		let sum = 0;
		for (let i = 1; i <= this.n; i++) sum += Math.pow(i, 2);
		return sum;
	}

	get squareOfSum() {
		let sum = 0;
		for (let i = 1; i <= this.n; i++) sum += i;
		return Math.pow(sum, 2);
	}

	get difference() {
		return this.squareOfSum - this.sumOfSquares;
	}
}

File: solutions/dart/difference-of-squares/1/lib/difference_of_squares.dart
import 'dart:math';

class DifferenceOfSquares {
  int squareOfSum(int n) {
    int total = 0;
    for (var i = 1; i <= n; i++) {
      total += i;
    }
    return pow(total, 2).toInt();
  }

  int sumOfSquares(int n) {
    int total = 0;
    for (var i = 1; i <= n; i++) {
      total += pow(i, 2).toInt();
    }
    return total;
  }

  int differenceOfSquares(int n) {
    return squareOfSum(n) - sumOfSquares(n);
  }
}
